### PR TITLE
Tao 6174/csv import

### PR DESCRIPTION
**Pull request by ionutpad**
Requires:
- [x] https://github.com/oat-sa/tao-core/pull/1713

### CODE REVIEW

File: controller/Import.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        $myForm = $formContainer->getForm();
+
+        //if the form is submited and valid
+        if($myForm->isSubmited()){
+            if($myForm->isValid()){
+                $options = $myForm->getValues();
+
+                /** @var UploadService $uploadService */
+                $uploadService = $this->getServiceLocator()->get(UploadService::SERVICE_ID);
+                $filePath = $uploadService->getUploadedFile($options['importFile']);
```
Comment (siwane) [inline controller/Import.php]: This method is deprecated, use getFlyFileSystem to fetch directly a File object

File: model/import/TestCenterCsvImporterFactory.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+     *
+     * @param $type
+     * @return TestCenterImportServiceInterface
+     * @throws \common_exception_NotFound
+     * @throws InvalidService
+     * @throws InvalidServiceManagerException
+     */
+    public function getImporter($type)
+    {
+        $typeOptions = $this->getOption(self::OPTION_MAPPERS);
```
Comment (siwane) [inline model/import/TestCenterCsvImporterFactory.php:43]: Could u take an eyes to `\oat\oatbox\service\ConfigurableService::getSubService // buildService`, it construct and load ConfigurableService. The con is that your objects are not ConfigurableService

File: model/import/RdsTestCenterImportService.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+     */
+    protected function persist(ImportMapper $mapper)
+    {
+        if (!$mapper instanceof TestCenterMapper) {
+            throw new \Exception('Mapper should be a TestCenterMapper');
+        }
+
+        $properties = $mapper->getProperties();
+        $class = $this->getTestCenterClass($properties);
+        $resource = $class->createInstanceWithProperties($properties);
```
Comment (siwane) [inline model/import/RdsTestCenterImportService.php:55]: Can u filter the properties to control what we store in database.

File: scripts/update/Updater.php [hunk 4]
```diff
@@ -127,5 +136,31 @@ public function update($initialVersion)
 
             $this->setVersion('3.9.0');
         }
+
+        if ($this->isVersion('3.9.0')) {
```
Comment (siwane) [inline scripts/update/Updater.php:140]: don"t need to bump two time the version

Comment (siwane) [reply scripts/update/Updater.php:140]: The manfiest bump is missing

**Review by siwane — CHANGES_REQUESTED** [review]

File: model/import/RdsTestCenterImportService.php (hunk 3 above)
Comment (ionutpad) [reply model/import/RdsTestCenterImportService.php:55]: Then it's no point in the mapper dynamic properties. This properties are the one configurable in the import, if you don't want a property to be saved you can filter out

File: scripts/update/Updater.php (hunk 4 above)
Comment (ionutpad) [reply scripts/update/Updater.php:140]: Adjusting, btw the manifest was not missing

File: model/import/TestCenterCsvImporterFactory.php (hunk 2 above)
Comment (ionutpad) [reply model/import/TestCenterCsvImporterFactory.php]: I will use getSubService I didn't know about this, the difference that getSubService offers lazy loading and thrown Exception?

Comment (ionutpad) [reply model/import/TestCenterCsvImporterFactory.php:43]: You had the same approach in the UserCsvImporterFactory :)

File: scripts/update/Updater.php [hunk 5]
```diff
@@ -127,5 +136,28 @@ public function update($initialVersion)
...
+
+            $service = new TestCenterCsvImporterFactory(array(
+                TestCenterCsvImporterFactory::OPTION_DEFAULT_SCHEMA => array(
+                    ImportMapper::OPTION_SCHEMA_MANDATORY => [
+                        'label' => OntologyRdfs::RDFS_LABEL,
+                    ],
+                    ImportMapper::OPTION_SCHEMA_OPTIONAL => []
+                )
+            ));
+            $typeOptions = $service->getOption(TestCenterCsvImporterFactory::OPTION_MAPPERS);
```
Comment (siwane) [inline scripts/update/Updater.php]: don't need to fetch mappers, the service does not exist yet

**Pull request merged**